src/client/power-pages/actions-hub/ActionsHubCommandHandlers.ts

```diff
@@ -259,3 +265,98 @@ export const uploadSite = async (siteTreeItem: SiteTreeItem) => {
     const modelVersion = siteTreeItem.siteInfo.dataModelVersion;
     PacTerminal.getTerminal().sendText(`pac pages upload --path "${websitePath}" --modelVersion "${modelVersion}"`);
 }
+
+/**
+ * Finds Power Pages sites in the parent folder that aren't in the known sites list
+ * @param knownSiteIds Map of site IDs that should be excluded from results
+ * @returns Array of site information objects for sites found in the parent folder
+ */
+export function findOtherSites(knownSiteIds: Map<string, boolean>, fsModule = fs, yamlModule = yaml): IOtherSiteInfo[] {
+    // Get the workspace folders
+    const workspaceFolders = vscode.workspace.workspaceFolders;
+    if (!workspaceFolders || workspaceFolders.length === 0) {
+        return [];
+    }
+
+    const currentWorkspaceFolder = workspaceFolders[0].uri.fsPath;
+    const parentFolder = path.dirname(currentWorkspaceFolder);
+
+    try {
+        // Get directories in the parent folder
+        const items = fsModule.readdirSync(parentFolder, { withFileTypes: true });
+        const directories = items
+            .filter(item => item.isDirectory())
+            .map(item => path.join(parentFolder, item.name));
+
+        // Make sure we include the current workspace folder
+        if (!directories.includes(currentWorkspaceFolder)) {
+            directories.push(currentWorkspaceFolder);
+        }
+
+        const otherSites: IOtherSiteInfo[] = [];
+
+        // Check each directory for website.yml
+        for (const dir of directories) {
+            const websiteYamlPath = path.join(dir, WEBSITE_YML);
+
+            if (fsModule.existsSync(websiteYamlPath)) {
+                try {
+                    // Parse website.yml to get site details
+                    const yamlContent = fsModule.readFileSync(websiteYamlPath, UTF8_ENCODING);
+                    const websiteData = yamlModule.load(yamlContent) as WebsiteYaml;
+
+                    const websiteId = websiteData?.adx_websiteid;
```

Comment:
I think we should be able to replace this with `getWebsiteRecordId` from `WorkspaceInfoFinderUtil.ts`